apps/crn-server/src/routes/research-outputs.route.ts

```diff
@@ -45,25 +50,31 @@ export const researchOutputRouteFactory = (
   );
 
   researchOutputRoutes.post('/research-outputs', async (req, res) => {
-    const { body, loggedInUser } = req;
+    const { body, loggedInUser, query } = req;
     const createRequest = validateResearchOutputPostRequestParameters(body);
     validateResearchOutputPostRequestParametersIdentifiers(createRequest);
 
-    if (
-      !loggedInUser ||
-      !hasCreateUpdateResearchOutputPermissions(
-        loggedInUser,
-        createRequest.teams,
-      )
-    ) {
+    const userRole = getUserRole(
```

Comment:
any reason why we need to get the role first only to pass it on the `hasEditResearchOutputPermission`? Could it not accept the logged in user and teams? is it used elsewhere where these params are not available?

Reply:
no, but in my opinion having `getUserRole` separated is more organized. Also here
 
<img width="847" alt="Screenshot 2023-03-03 at 09 02 11" src="https://user-images.githubusercontent.com/16595804/222715495-e44e9dee-4b76-4f95-ae3c-77ce98fed664.png">

if we had `getUserRole` as part of each function, it would be called 3 times instead of 1. This wouldn't make a difference, but I don't think we gain a lot moving `getUserRole` inside each function as well ¯\_(ツ)_/¯

Reply:
well i think its separation of concerns really. Once again having to determine the role by the client and then having to send it across to the permission service means shifting that responsibility onto the clients and they become aware of our role-based system. I realise some architectural patterns may seem less performant but this could haunt us in the future so I was trying to avoid it. But it's fine, if you feel it's better this way lets leave it for now